AutoCollection/HttpDependencies.ts

```diff
@@ -122,31 +122,40 @@ class AutoCollectHttpDependencies {
             if (client.config && client.config.correlationId) {
-                const correlationHeader = telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader);
-                if (correlationHeader) {
-                    const components = correlationHeader.split(",");
-                    const key = `${RequestResponseHeaders.requestContextSourceKey}=`;
-                    if (!components.some((value) => value.substring(0,key.length) === key)) {
-                        telemetry.request.setHeader(
-                            RequestResponseHeaders.requestContextHeader,
-                            `${correlationHeader},${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
+                // getHeader returns "any" type in newer versions of node. In basic scenarios, this will be <string | string[] | number>, but could be modified to anything else via middleware
+                const correlationHeader = <any>telemetry.request.getHeader(RequestResponseHeaders.requestContextHeader)
+                let header: string; // cast correlationHeader to string
+                if (typeof correlationHeader === "string") {
+                    header = correlationHeader;
+                } else if (correlationHeader instanceof Array) { // string[]
+                    header = correlationHeader.join(",");
+                } else if (correlationHeader && typeof (correlationHeader as any).toString === "function") {
+                    // best effort attempt: requires well-defined toString
+                    try {
+                        header = (correlationHeader as any).toString();
+                    } catch (err) {
+                        Logging.warn("Outgoing request-context header could not be parsed. Correlation of requests may be lost.", err, correlationHeader);
                     }
+                }
+
+                if (header) {
+                    AutoCollectHttpDependencies.addRequestCorrelationIdHeaderFromString(client, telemetry, header);
                 } else {
                     telemetry.request.setHeader(
                         RequestResponseHeaders.requestContextHeader,
                         `${RequestResponseHeaders.requestContextSourceKey}=${client.config.correlationId}`);
                 }
-            }
 
-            if (currentContext && currentContext.operation) {
-                telemetry.request.setHeader(RequestResponseHeaders.requestIdHeader, uniqueRequestId);
-                // Also set legacy headers
-                telemetry.request.setHeader(RequestResponseHeaders.parentIdHeader, currentContext.operation.id);
-                telemetry.request.setHeader(RequestResponseHeaders.rootIdHeader, uniqueRequestId);
+                if (currentContext && currentContext.operation) {
+                    telemetry.request.setHeader(RequestResponseHeaders.requestIdHeader, uniqueRequestId);
```

Comment:
Don't these need try-catch?

Reply:
Added try/catch for setHeader group in outgoing requests